File: PipelineService/Models/MqttMessages/FrontendExecutionNotification.cs
using System;
using System.Collections.Generic;
using PipelineService.Models.Pipeline;

namespace PipelineService.Models.MqttMessages
{
	public class FrontendExecutionNotification : BaseMqttMessage
	{
		public Guid PipelineId { get; set; }
		public Guid ExecutionId { get; set; }
		public Guid OperationId { get; set; }
		public bool Successful { get; set; }
		public bool Cached { get; set; }
		public int ExecutionTime { get; set; }
		public string ErrorDescription { get; set; }
		public int OperationsExecuted { get; set; }
		public int OperationsInExecution { get; set; }
		public int OperationsToBeExecuted { get; set; }
		public int OperationsFailedToExecute { get; set; }
		public string OperationName { get; set; }
		public DateTime CompletedAt { get; set; }
		public IList<Dataset> ResultDatasets { get; set; }
	}
}

File: DevFrontend/src/app/pipeline/_model/frontend-execution-notification.ts
import {Dataset} from './dataset';

export interface FrontendExecutionNotification {
	PipelineId: string;
	ExecutionId: string;
	OperationId: string;
	OperationName: string;
	Successful: boolean;
	Cached: boolean;
	ExecutionTime: any;
	CompletedAt: Date;
	ErrorDescription: string;
	OperationsExecuted: number;
	OperationsInExecution: number;
	OperationsToBeExecuted: number;
	OperationsFailedToExecute: number;
	ResultDatasets: Array<Dataset>;
}
